toolkit/smart-contracts/plutus-data/src/key_value.rs

```diff
@@ -0,0 +1,82 @@
+use crate::{DataDecodingError, DecodingResult};
+use cardano_serialization_lib::{PlutusData, PlutusList};
+use sidechain_domain::byte_string::ByteString;
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct KeyValueDatum {
+	pub key: String,
+	pub value: ByteString,
+}
+
+impl TryFrom<PlutusData> for KeyValueDatum {
+	type Error = DataDecodingError;
+	fn try_from(datum: PlutusData) -> DecodingResult<Self> {
+		let data_iter = datum.as_list().ok_or_else(|| DataDecodingError {
+			datum: datum.clone(),
+			to: "KeyValueDatum".into(),
+			msg: "Expected [String, Vec<u8>]".into(),
```

Comment:
Cosmetic: If the error is same in all places, then please chain with `.and_then` and call ok_or_else at the end.
Also, "Expected List([Bytes, Bytes)]". We don't know here if it can be made to string and also Vec<u8> is a Rust concept, but we deal with Plutus Data here.